### PR TITLE
Migrate tink-server to cobra and viper

### PR DESCRIPTION
**Pull request by gianarb**
This PR refactors how tink-server starts.
It uses Cobra and Viper same as tink-cli and tink-worker.

Right now it tries to keep compatibility with the old way of doing
things.

TODO:

- [x] run this version in sandbox without any change. it should work like before
- [ ] ~Unit test (project aborted, I don't know how to mock env var!)~

Signed-off-by: Gianluca Arbezzano <gianarb92@gmail.com>

### CODE REVIEW

**Comment by gianarb** [comment]
@displague you did a similar job for the other binaries. Can you have a look and tell me if it makes sense? I know you don't like to keep "useless retro-compatibility" but the code will get removed soon trust me!! 😇 

Thanks

**Comment by gianarb** [comment]
@detiber I know you have this PR #266 in the fight but let's hold (it is pretty stale already) it until we figure out #431

File: cmd/tink-server/main.go [hunk 1]
```diff
@@ -22,89 +27,246 @@ var (
...
-	tinkDB := db.Connect(dbCon, logger)
-
-	_, onlyMigration := os.LookupEnv("ONLY_MIGRATION")
-	if onlyMigration {
-		logger.Info("Applying migrations. This process will end when migrations will take place.")
-		numAppliedMigrations, err := tinkDB.Migrate()
-		if err != nil {
-			log.Fatal(err)
-			panic(err)
+	if pguser := os.Getenv("PGUSER"); pguser != "" {
```
Comment (displague) [inline cmd/tink-server/main.go:67]: Any reason we aren't using automatic environment variable handling?
https://github.com/tinkerbell/tink/blob/0b477fd65ffc64c1143c44c3799d2c1bc39d7437/cmd/tink-worker/cmd/root.go#L115-L121

Comment (displague) [reply cmd/tink-server/main.go:67]: I see.. https://github.com/tinkerbell/tink/pull/431/files#diff-d33013e31f0cb0caf39aaef7185eec15145f19bdf8da9e32b63b37d669572f3bR128-R133

**Review by mmlb — APPROVED** [review]

**Comment by gianarb** [comment]
```
$ go run cmd/tink-server/main.go --help
Usage:
  tink-server [flags]

Flags:
      --cert-dir string
      --facility string            This is temporary. It will be removed (default "deprecated")
      --grpc-authority string      The address used to expose the gRPC server (default ":42113")
  -h, --help                       help for tink-server
      --http-authority string      The address used to expose the HTTP server (default ":42114")
      --only-migration             When enabled it
      --postgres-database string   The Postgres database name (default "tinkerbell")
      --postgres-password string   The Postgres database password (default "tinkerbell")
      --postgres-sslmode string    Enable or disable SSL mode in postgres (default "disable")
      --postgres-user string       The Postgres database username (default "tinkerbell")
      --tls-cert string
```

**Review by mmlb — DISMISSED** [review]

**Review by mmlb — APPROVED** [review]

**Pull request merged**